pkg/ui/src/views/queries/statementDetails.tsx

```diff
@@ -0,0 +1,209 @@
+import { FixLong } from "src/util/fixLong";
+import { Duration } from "src/util/format";
+import { SqlBox } from "src/views/shared/components/sql/box";
+import { SummaryBar, SummaryHeadlineStat } from "src/views/shared/components/summaryBar";
+import * as protos from "src/js/protos";
+
+import { countBarChart, latencyBarChart } from "./barCharts";
+
+type StatementStatistics = protos.cockroach.sql.CollectedStatementStatistics$Properties;
+
+interface StatementDetailsOwnProps {
+  statement: StatementStatistics;
+  refreshQueries: typeof refreshQueries;
+}
+
+type StatementDetailsProps = StatementDetailsOwnProps & RouterState;
+
+interface NumericStat {
+  mean?: number;
+  squared_diffs?: number;
+}
+
+function variance(stat: NumericStat, count: number) {
+  return stat.squared_diffs / (count - 1);
+}
+
+interface NumericStatRow {
+  name: string;
+  value: NumericStat;
+}
+
+interface NumericStatTableProps {
+  rows: NumericStatRow[];
+  count: number;
+  format: (v: number) => string;
+}
+
+class NumericStatTable extends React.Component<NumericStatTableProps> {
+  static defaultProps = {
+    format: (v: number) => `${v}`,
+  };
+
+  render() {
+    return (
+      <table className="numeric-stats-table">
+        <thead>
+          <tr className="numeric-stats-table__row--header">
+            <th className="numeric-stats-table__cell" />
+            <th className="numeric-stats-table__cell">Mean</th>
+            <th className="numeric-stats-table__cell">Variance</th>
+          </tr>
+        </thead>
+        <tbody style={{ textAlign: "right" }}>
+          {
+            this.props.rows.map((row: NumericStatRow) => {
+              return (
+                <tr className="numeric-stats-table__row--body">
+                  <th className="numeric-stats-table__cell" style={{ textAlign: "left" }}>{ row.name }</th>
+                  <td className="numeric-stats-table__cell">{ this.props.format(row.value.mean) }</td>
+                  <td className="numeric-stats-table__cell">{ this.props.format(variance(row.value, this.props.count)) }</td>
+                </tr>
+              );
+            })
+          }
+        </tbody>
+      </table>
+    );
+  }
+}
+
+class StatementDetails extends React.Component<StatementDetailsProps> {
+  componentWillMount() {
+    this.props.refreshQueries();
+  }
+
+  componentWillReceiveProps() {
+    this.props.refreshQueries();
+  }
+
+  render() {
+    if (!this.props.statement) {
+      return "loading...";
+    }
+
+    const { stats } = this.props.statement;
+
+    const count = FixLong(stats.count).toInt();
+    const firstAttemptCount = FixLong(stats.first_attempt_count).toInt();
+
+    const countBar = countBarChart();
+    const latencyBar = latencyBarChart();
+
+    return (
+      <div>
+        <Helmet>
+          <title>{`Details | Statements`}</title>
+        </Helmet>
+        <section className="section">
+          <h2>Statement Details</h2>
```

Comment:
Hate to bikeshed on terminology, but maybe rename the main page to "statements" too? Meh, I dunno.

Also, maybe "Statement Fingerprint Details"? Probably want to reinforce that this is a set of queries with the same fingerprint, not a single query.

Reply:
"Statements" is definitely growing on me. Seems unambiguous and non-confusing of all options.

Reply:
Done.